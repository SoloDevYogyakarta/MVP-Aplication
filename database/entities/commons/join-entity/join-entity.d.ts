import { Model, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
export declare class JoinEntity extends BaseEntity {
    source_id: string;
    foreign_id: string;
}
export type JoinCreationAttribute = Optional<JoinEntity, 'id'>;
export interface JoinInstance extends Model<JoinCreationAttribute, JoinEntity>, JoinEntity {
}
export declare const joinEntity: import("sequelize").ModelCtor<JoinInstance>;
