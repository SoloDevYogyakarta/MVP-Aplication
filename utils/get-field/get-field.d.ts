import { GetFieldType } from 'src/types/get-field.type';
export declare function getField(filename?: GetFieldType): any;
