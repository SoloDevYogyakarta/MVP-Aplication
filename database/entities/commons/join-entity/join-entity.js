"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.joinEntity = exports.JoinEntity = void 0;
const nanoid_1 = require("nanoid");
const sequelize_1 = require("sequelize");
const base_entity_1 = require("../../base-entity/base-entity");
const entity_1 = require("../../entity");
class JoinEntity extends base_entity_1.BaseEntity {
}
exports.JoinEntity = JoinEntity;
exports.joinEntity = entity_1.sequelize.define('JOIN', {
    public_id: sequelize_1.DataTypes.STRING,
    source_id: sequelize_1.DataTypes.STRING,
    foreign_id: sequelize_1.DataTypes.STRING,
}, {
    tableName: 'JOIN',
    schema: 'COMMONS',
    timestamps: false,
    hooks: {
        beforeCreate(attributes, options) {
            const instance = attributes;
            instance.public_id = (0, nanoid_1.nanoid)();
        },
    },
});
//# sourceMappingURL=join-entity.js.map