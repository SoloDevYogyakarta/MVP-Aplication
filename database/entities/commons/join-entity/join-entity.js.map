{"version":3,"file":"join-entity.js","sourceRoot":"","sources":["../../../../../src/database/entities/commons/join-entity/join-entity.ts"],"names":[],"mappings":";;;AAAA,mCAAgC;AAChC,yCAAwE;AACxE,+DAA2D;AAC3D,yCAAyC;AAEzC,MAAa,UAAW,SAAQ,wBAAU;CAGzC;AAHD,gCAGC;AAMY,QAAA,UAAU,GAAG,kBAAS,CAAC,MAAM,CACxC,MAAM,EACN;IACE,SAAS,EAAE,qBAAS,CAAC,MAAM;IAC3B,SAAS,EAAE,qBAAS,CAAC,MAAM;IAC3B,UAAU,EAAE,qBAAS,CAAC,MAAM;CACI,EAClC;IACE,SAAS,EAAE,MAAM;IACjB,MAAM,EAAE,SAAS;IACjB,UAAU,EAAE,KAAK;IACjB,KAAK,EAAE;QACL,YAAY,CAAC,UAAU,EAAE,OAAO;YAC9B,MAAM,QAAQ,GAAG,UAAU,CAAC;YAC5B,QAAQ,CAAC,SAAS,GAAG,IAAA,eAAM,GAAE,CAAC;QAChC,CAAC;KACF;CACF,CACF,CAAC"}