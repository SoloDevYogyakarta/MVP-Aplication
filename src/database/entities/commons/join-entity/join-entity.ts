import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class JoinEntity extends BaseEntity {
  source_id: string;
  foreign_id: string;
}

export type JoinCreationAttribute = Optional<JoinEntity, 'id'>;
export interface JoinInstance
  extends Model<JoinCreationAttribute, JoinEntity>,
    JoinEntity {}
export const joinEntity = sequelize.define<JoinInstance>(
  'JOIN',
  {
    public_id: DataTypes.STRING,
    source_id: DataTypes.STRING,
    foreign_id: DataTypes.STRING,
  } as ModelAttributes<JoinInstance>,
  {
    tableName: 'JOIN',
    schema: 'COMMONS',
    timestamps: false,
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
      },
    },
  },
);
