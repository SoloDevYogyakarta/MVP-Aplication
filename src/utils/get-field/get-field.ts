import { GetFieldType } from 'src/types/get-field.type';
import { readpath } from '../system/system';

export function getField(filename?: GetFieldType) {
  try {
    if (filename === 'fake') {
      return readpath(`../../database/dataTxt/${filename}.txt`);
    }
    return JSON.parse(readpath(`../../database/dataTxt/${filename}.txt`));
  } catch (err) {
    throw new Error(err);
  }
}
