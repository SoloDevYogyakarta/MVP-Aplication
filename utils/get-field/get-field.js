"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getField = void 0;
const system_1 = require("../system/system");
function getField(filename) {
    try {
        const inc = ['token', 'fake'];
        if (inc.includes(filename)) {
            return (0, system_1.readpath)(`../../database/dataTxt/${filename}.txt`);
        }
        return JSON.parse((0, system_1.readpath)(`../../database/dataTxt/${filename}.txt`));
    }
    catch (err) {
        throw new Error(err);
    }
}
exports.getField = getField;
//# sourceMappingURL=get-field.js.map