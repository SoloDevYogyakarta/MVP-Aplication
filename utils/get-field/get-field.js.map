{"version":3,"file":"get-field.js","sourceRoot":"","sources":["../../../src/utils/get-field/get-field.ts"],"names":[],"mappings":";;;AACA,6CAA4C;AAE5C,SAAgB,QAAQ,CAAC,QAAuB;IAC9C,IAAI;QACF,MAAM,GAAG,GAAG;YACV,MAAM;YACN,OAAO;YACP,cAAc;YACd,cAAc;YACd,gBAAgB;SACjB,CAAC;QACF,IAAI,GAAG,CAAC,QAAQ,CAAC,QAAQ,CAAC,EAAE;YAC1B,OAAO,IAAA,iBAAQ,EAAC,kBAAkB,QAAQ,MAAM,CAAC,CAAC;SACnD;QACD,OAAO,IAAI,CAAC,KAAK,CAAC,IAAA,iBAAQ,EAAC,kBAAkB,QAAQ,MAAM,CAAC,CAAC,CAAC;KAC/D;IAAC,OAAO,GAAG,EAAE;QACZ,MAAM,IAAI,KAAK,CAAC,GAAG,CAAC,CAAC;KACtB;AACH,CAAC;AAhBD,4BAgBC"}