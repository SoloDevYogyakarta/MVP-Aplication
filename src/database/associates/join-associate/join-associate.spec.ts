import { ModelCtor } from 'sequelize';
import { getField } from '../../../utils/get-field/get-field';
import { JoinInstance } from '../../../database/entities/commons/join-entity/join-entity';
import {
  joinProductAndFileAssociate,
  joinProductAndFileAttribute,
  joinProductAndFileInclude,
} from './join-associate';

describe('JoinAssociate', () => {
  let entity: ModelCtor<JoinInstance>;

  beforeEach(() => {
    entity = joinProductAndFileAssociate;
  });

  it('should to be defined', () => expect(entity).toBeDefined());

  it('render correctly', () => expect(entity).toMatchSnapshot());

  try {
    it('findOne with relationship', async () => {
      const { public_id } = getField('basic-http-entity');
      const findOne = await entity.findOne({
        where: { source_id: public_id },
        attributes: joinProductAndFileAttribute,
        include: joinProductAndFileInclude,
      });
      expect(findOne.source_id).toEqual(public_id);
    });

    it('findAll with relationship', async () => {
      const findAll = await entity.findAll({
        attributes: joinProductAndFileAttribute,
        include: joinProductAndFileInclude,
      });
      expect(findAll.length).not.toEqual(0);
    });
  } catch (err) {
    // empty
  }
});
