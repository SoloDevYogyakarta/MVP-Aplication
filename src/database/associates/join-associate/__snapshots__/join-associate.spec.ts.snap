// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinAssociate render correctly 1`] = `[Function]`;
